Pc kp user report job

pconrad opened the pull request:
In this PR, we create an option for the admin to download a User Report.

First, we create a job that can be scheduled via the Heroku Scheduler:
* `bin/newUserReport` - this iterates over all of the commons, and for each commons, creates a new report for that commons

Second, we add to the admin page, a list of all currently available reports.
* Each of these has a clickable link to a show page for that report that shows the detail of that report
* The detail has the same information that shows for the first 15 users in the commons on the current admin page
* Each report has a delete button; the delete button deletes the report, and all of it's associated data
* When you go to the show page, there is a CSV download button so that you can download that report.

![image](https://user-images.githubusercontent.com/1119017/115096501-41dd1880-9eda-11eb-9398-8005d8d7c59c.png)

![image](https://user-images.githubusercontent.com/1119017/115096512-4dc8da80-9eda-11eb-8993-b5de13235d85.png)


The intended workflow is that the admin will download reports as needed, and then delete them so that they do not
clutter up the database.

# Future work

We may want to create a job that deletes reports that are older than some interval (e.g. 14 days), to prevent the database from accumulating too many stale reports.   We might also add this logic into the job that creates the reports in the first place; i.e. we can set a limit on how many reports may exist at any one time (say 10?), and we can make it automatic to delete all but the most recent 10 reports.

kevbbn reviewed (commented):
I don't really know the sequelizer syntax but it looks like you just added a function to get a userid from SQL using the sequelizer library, and the HTML added is just the new sections at the top of the home page for admins. If this works on your local PC then it looks pretty good in my opinion

sethvanb commented on views/user_choose_commons.js:
```diff
@@ -1,14 +1,17 @@
     const commons = await get_all_commons(req);
     const commons_user = await get_user_commons(req, String(user_obj.id));
+    const user_info = await get_user_info(String(user_obj.id));
+    console.log("user_info=",user_info);
```
Do we want a console.log being merged?

sethvanb reviewed (commented):
Other than the one console.log the code looks fine. I do not know sql tho so the query is not 100% confirmed to be efficient.

sethvanb reviewed (changes requested):
Take out console.logs

pconrad committed:
pc/kp - define database tables for user reports

pconrad committed:
pc/kp - add a script for a cron job to generate user reports

This user report is put into a two new database tables,
CommonsUserReports and CommonsUserReportItems.
In the next few commits, we'll build the UI
that allows the user to pull up those reports and
view them, as well as deleting them.

pconrad committed:
pc/kp --- add user interface for reports
